compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala

```diff
@@ -270,7 +270,7 @@ class ReifyQuotes extends MacroTransform {
 
       val tpe = MethodType(defn.SeqType.appliedTo(defn.AnyType) :: Nil, tree.tpe.widen)
       val meth = ctx.newSymbol(lambdaOwner, UniqueName.fresh(nme.ANON_FUN), Synthetic | Method, tpe)
-      Closure(meth, tss => body(tss.head.head)(ctx.withOwner(meth)).changeOwner(ctx.owner, meth)).withSpan(tree.span)
+      Closure(meth, tss => body(tss.head.head)(ctx.withOwner(meth)).changeNonLocalOwners(meth)).withSpan(tree.span)
```

Comment:
@nicolasstucki I'm not sure if this is the best fix. Could you advise? The detailed message is in 235e393965bc9680f0d0c1a386f9d7c87a6b22d0.

Reply:
This change seems to be good